### PR TITLE
chore(self-managed): set index page for supported platforms

### PR DESCRIPTION
**Pull request by aabouzaid**
## What is the purpose of the change

Adding an index page for the `platforms` category where it has its own link as well as embedded auto-generated items. 

This will make it better for readability as well when we want to refer/share a specific category.

Here is the new introduced link (I will add it for 8.1 when the review is done).
https://docs.camunda.io/docs/next/self-managed/platform-deployment/helm-kubernetes/platforms/

And here is a screenshot for that new index page:
c8sm-k8s-platforms

## Are there related marketing activities

No.

## When should this change go live?

No specific time, as soon as possible.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory.
- [x] My changes do not require an Engineering review.
- [x] My changes do not require a technical writer review.

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -648,11 +648,19 @@ module.exports = {
             "self-managed/platform-deployment/helm-kubernetes/deploy",
             "self-managed/platform-deployment/helm-kubernetes/upgrade",
             {
-              Platforms: [
+              type: "category",
+              label: "Platforms",
+              link: {
+                type: "doc",
+                id: "self-managed/platform-deployment/helm-kubernetes/platforms/platforms",
```
Comment (aabouzaid) [inline sidebars.js:655]: Regarding `platforms/platforms`, I'm using the same convention mentioned in the official docs for [Category index](https://docusaurus.io/docs/2.0.1/sidebar/autogenerated#category-index-convention).

**Review by pepopowitz — APPROVED** [review]
From a docs infrastructure perspective, I think this looks great! 

I defer to @christinaausley in regards to content & information architecture.

File: docs/self-managed/platform-deployment/helm-kubernetes/platforms/platforms.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+id: platforms
+title: "Camunda Platform 8 officially supported Kubernetes platforms"
```
Comment (christinaausley) [inline docs/self-managed/platform-deployment/helm-kubernetes/platforms/platforms.md:3]: I'd ideally like to see a shorter title -- perhaps "Supported Kubernetes platforms"?

File: docs/self-managed/platform-deployment/helm-kubernetes/platforms/platforms.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+description: "An overview of Camunda Platform 8 officially supported Kubernetes platforms"
+---
+
+import DocCardList from '@theme/DocCardList';
+import {useCurrentSidebarCategory} from '@docusaurus/theme-common';
+
+[Deploying Camunda Platform 8 using Helm charts](../deploy.md) will work for all platforms and distributions. However, each platform or cloud provider may have some special prerequisites or pitfalls to avoid. The platforms guides highlights important notes for a smooth Camunda Platform deployment on various platforms.
```
Comment (christinaausley) [inline docs/self-managed/platform-deployment/helm-kubernetes/platforms/platforms.md:10]: "The platforms guides highlights important notes for a smooth Camunda Platform deployment on various platforms."

Do you mean "The Platform guides highlight important notes..."? Could we link to what you've mentioned here?

**Review by christinaausley — CHANGES_REQUESTED** [review]
A few comments for clarification and some adjustments to the wording in my latest commit -- let me know if you have any questions about these.

File: docs/self-managed/platform-deployment/helm-kubernetes/platforms/platforms.md (hunk 3 above)
Comment (aabouzaid) [reply docs/self-managed/platform-deployment/helm-kubernetes/platforms/platforms.md]: @christinaausley I've rephrased it. I was referencing to "Kubernetes platform" not the "Camunda Platform".

Regarding the links, they are auto-generated at the end of the page, as shown in this screenshot:
201646460-94ee85f6-23e2-49f1-b999-95a21aad5289

**Comment by aabouzaid** [comment]
@pepopowitz thanks for the review :+1: 

@christinaausley I've addressed your comments but by mistake, I overrode your "commit" :see_no_evil: 
so could you please push it again? :bow:

**Review by christinaausley — APPROVED** [review]

**Comment by aabouzaid** [comment]
This PR is ready to be squashed and merged :heavy_check_mark: (github removed the previous approval)

**Review by christinaausley — APPROVED** [review]

**Pull request merged**